ttnn/cpp/ttnn/operations/pool/generic/device/pool_multi_core_program_factory.cpp

```diff
@@ -232,59 +319,55 @@ Pool2D::MultiCore::cached_program_t pool2d_multi_core_sharded_with_halo_v2_impl_
-        log_debug(
-            tt::LogOp,
-            "in_reader_indices_cb :: PS = {}, NP = {}",
-            in_reader_indices_cb_pagesize,
-            in_reader_indices_cb_npages);
-        log_debug(tt::LogOp, "in_scalar_cb :: PS = {}, NP = {}", in_scalar_cb_pagesize, in_scalar_cb_npages);
-        log_debug(tt::LogOp, "out_cb :: PS = {}, NP = {}", out_cb_pagesize, out_cb_npages);
-        log_debug(tt::LogOp, "in_addr: {}", src_dram_buffer->address());
-        log_debug(tt::LogOp, "in_reader_indices_addr: {}", reader_indices_storage.get_buffer()->address());
-        log_debug(tt::LogOp, "out_addr: {}", dst_dram_buffer->address());
-        log_debug(tt::LogOp, "kernel_size_h: {}", kernel_size_h);
-        log_debug(tt::LogOp, "kernel_size_w: {}", kernel_size_w);
-        log_debug(tt::LogOp, "kernel_size_hw: {}", kernel_size_hw);
-        log_debug(tt::LogOp, "kernel_size_hw_padded: {}", kernel_size_hw_padded);
-        log_debug(tt::LogOp, "stride_h: {}", stride_h);
-        log_debug(tt::LogOp, "stride_w: {}", stride_w);
-        log_debug(tt::LogOp, "pad_h: {}", pad_h);
-        log_debug(tt::LogOp, "pad_w: {}", pad_w);
-        log_debug(tt::LogOp, "out_h: {}", out_h);
-        log_debug(tt::LogOp, "out_w: {}", out_w);
-        log_debug(tt::LogOp, "out_w_loop_count: {}", out_w_loop_count);
-        log_debug(tt::LogOp, "out_c: {}", output_shape[3]);
-        log_debug(tt::LogOp, "out_nbytes_c: {}", out_nbytes_c);
-        log_debug(tt::LogOp, "in_h: {}", in_h);
-        log_debug(tt::LogOp, "in_w: {}", in_w);
-        log_debug(tt::LogOp, "in_c: {}", input_shape[3]);
-        log_debug(tt::LogOp, "in_ntiles_c: {}", in_ntiles_c);
-        log_debug(tt::LogOp, "in_nblocks_c: {}", in_nblocks_c);
-        log_debug(tt::LogOp, "in_nbytes_c: {}", in_nbytes_c);
-        log_debug(tt::LogOp, "out_ntiles_c: {}", out_ntiles_c);
-        log_debug(tt::LogOp, "nblocks: {}", nblocks);
-        log_debug(tt::LogOp, "ncores: {}", ncores);
-        log_debug(tt::LogOp, "in_nhw_per_core: {}", in_nhw_per_core);
-        log_debug(tt::LogOp, "out_nhw_per_core: {}", out_nhw_per_core);
-        log_debug(tt::LogOp, "split_reader: {}", split_reader);
-        log_debug(tt::LogOp, "multi_buffering_factor: {}", multi_buffering_factor);
-        log_debug(tt::LogOp, "is_wide_reduction: {}", is_wide_reduction);
-        log_debug(tt::LogOp, "is_large_kernel: {}", is_large_kernel);
-        log_debug(tt::LogOp, "is_in_sharded: {}", input.memory_config().is_sharded());
-        log_debug(tt::LogOp, "is_out_sharded: {}", output.memory_config().is_sharded());
-    }
-
     /**
      * Reader Kernel: input rows -> input cb
      */
-    uint32_t bf16_scalar = get_bf16_pool_scalar(pool_type, kernel_size_hw) << 16;
     float one = 1.;
     uint32_t bf16_one_u32 = *reinterpret_cast<uint32_t*>(&one);
+    uint32_t bf16_scalar = get_bf16_pool_scalar(pool_type, kernel_size_h, kernel_size_w, divisor_override);
```

Comment:
maybe put const/constexpr on couple of var declared here to make code more readable.